sql/src/main/java/io/crate/execution/engine/aggregation/GroupingCollector.java

```diff
@@ -181,7 +183,26 @@ private GroupingCollector(CollectExpression<Row, ?>[] expressions,
         return Collections.emptySet();
     }
 
-    private void onNextRow(Map<K, Object[]> statesByKey, Row row) {
+    private void reduce(Map<K, Object[]> statesByKey, Row row) {
+        for (CollectExpression<Row, ?> expression : expressions) {
+            expression.setNextRow(row);
+        }
+        K key = keyExtractor.apply(row);
+        Object[] states = statesByKey.get(key);
+        if (states == null) {
+            states = new Object[aggregations.length];
+            statesByKey.put(key, states);
+            for (int i = 0; i < aggregations.length; i++) {
+                states[i] = inputs[i][0].value();
+            }
+        } else {
+            for (int i = 0; i < aggregations.length; i++) {
+                states[i] = aggregations[i].reduce(ramAccountingContext, states[i], inputs[i][0].value());
+            }
+        }
```

Comment:
Shouldn't we add the calculation for the new entry in the map like here: https://github.com/crate/crate/pull/7545/files#diff-2f6039084f6fef0d16e87d1167f7129fR230?

Reply:
ah right, pushed a fixup